Detectors/MUON/MCH/Triggering/src/EventFinder.cxx

```diff
@@ -0,0 +1,120 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file EventFinder.cxx
+/// \brief Implementation of a class to group MCH digits based on MID information
+///
+/// \author Philippe Pillot, Subatech
+
+#include "MCHTriggering/EventFinder.h"
+
+#include <algorithm>
+#include <iterator>
+#include <utility>
+
+#include "CommonDataFormat/InteractionRecord.h"
+#include "MCHTriggering/EventFinderParam.h"
+
+namespace o2
+{
+namespace mch
+{
+
+//_________________________________________________________________________________________________
+/// run the event finder algorithm
+void EventFinder::run(const gsl::span<const mch::ROFRecord>& mchROFs,
```

Comment:
(note: this is nick picking on an otherwise very fine and well prepared PR ;-) ).
I would have split this run method in its actual 4 working parts to ease reading. 
(plus I suspect the digit merging part we'll have to externalise at some point to reuse it elsewhere...)
Again, not a call for action for this PR, just a comment.